### PR TITLE
cadl, use isApiVersion in cadl-dpg

### PR DESCRIPTION
**Pull request by haolingdong-msft**
WIP tests part.

### CODE REVIEW

File: cadl-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -242,7 +243,7 @@ export class CodeModelBuilder {
       server.parameters.forEach((it) => {
         let parameter;
 
-        if (it.name === "ApiVersion") {
+        if (isApiVersion(this.program, it as any)) {
```
Comment (haolingdong-msft) [inline cadl-extension/src/code-model-builder.ts:246]: This part I referred to [cadl-python](https://github.com/Azure/autorest.python/blob/main/packages/cadl-python/src/emitter.ts#L1143). I know that using `as any` is not a good practice, I saw Isabella created an issue in cadl-dpg, so I leave a question there https://github.com/Azure/cadl-azure/issues/2239#issuecomment-1398055201, I will keep our code updated with python-cadl.

Comment (weidongxu-microsoft) [reply cadl-extension/src/code-model-builder.ts:246]: You belongs to the cadl-dpg

I would assume we modify it to take `HttpOperationParameter | ModelProperty`
https://github.com/Azure/cadl-azure/blob/fb8b612/packages/cadl-dpg/src/utils.ts#L38-L42

**Review by weidongxu-microsoft — APPROVED** [review]

File: cadl-tests/cadl/server.cadl [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import "@cadl-lang/rest";
+import "@cadl-lang/versioning";
 
 using Cadl.Http;
+using Cadl.Versioning;
```
Comment (weidongxu-microsoft) [inline cadl-tests/cadl/server.cadl:5]: seems not need Cadl.Versioning

**Review by weidongxu-microsoft — APPROVED** [review]

File: cadl-tests/cadl/server.cadl [hunk 3]
```diff
@@ -15,10 +17,21 @@ using Cadl.Http;
 
     @doc("top-level domain, use org")
     tld?: string = "org",
+
+    @doc("Api Version")
+    @path
```
Comment (weidongxu-microsoft) [inline cadl-tests/cadl/server.cadl:22]: Do we need `@path`?

File: cadl-extension/src/code-model-builder.ts (hunk 1 above)
Comment (haolingdong-msft) [reply cadl-extension/src/code-model-builder.ts:246]: I can take this issue as well. I will merge this pr first.

**Review by XiaofeiCao — APPROVED** [review]

File: cadl-tests/cadl/apiversion.cadl [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+)
+namespace Cadl.ApiVersion;
+
+enum APIVersions {
+  v1_0: "v1.0",
+}
+
+@route("/apiversion")
+interface ApiVersionOp {
+  read(): string;
```
Comment (XiaofeiCao) [inline cadl-tests/cadl/apiversion.cadl:28]: What does this operation do? Should we can add a query parameter named `api-version`?

Comment (haolingdong-msft) [reply cadl-tests/cadl/apiversion.cadl:28]: We tested api version in path param. https://github.com/Azure/autorest.java/blob/c3d52e44fd85c31c835e3b176828b3117ca16a47/cadl-tests/src/main/java/com/cadl/apiversion/implementation/ApiVersionClientImpl.java#L140

**Pull request merged**